Client/src/Components/TabPanels/ProfileSettings/PasswordPanel.jsx

```diff
@@ -31,6 +32,76 @@ const PasswordPanel = () => {
   const [isLoading, setIsLoading] = useState(false);
   const [showPassword, setShowPassword] = useState(false);
 
+  //for testing, will tweak when I implement redux slice
+  const [localData, setLocalData] = useState({
+    "edit-current-password": {
+      value: "",
+      //TBD
+      type: "",
+    },
+    "edit-new-password": {
+      value: "",
+      type: "password",
+    },
+    "edit-confirm-password": {
+      value: "",
+      type: "confirm",
+    },
+  });
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (event) => {
+    const { value, id } = event.target;
+    setLocalData((prev) => ({
+      ...prev,
+      [id]: {
+        ...prev[id],
+        value: value,
+      },
+    }));
+    validateField(id, value);
+  };
+
+  const validateField = (id, value) => {
+    let error = "";
+    switch (localData[id].type) {
+      case "password":
+        error =
+          value.trim() === ""
+            ? "*This field is required."
+            : value.length < 8
+            ? "*Password must be at least 8 characters long."
+            : !/[A-Z]/.test(value)
+            ? "*Password must contain at least one uppercase letter."
+            : !/\d/.test(value)
+            ? "*Password must contain at least one number."
+            : !/[!@#$%^&*]/.test(value)
+            ? "*Password must contain at least one symbol."
+            : "";
```

Comment:
We use the [Joi](https://joi.dev/api/?v=17.13.3) library for field validation, and I believe we already have validation schemas set up for username and password.

Please see register/validation pages for implementaiton examples, it's pretty easy to use